src/js/utils.js

```diff
@@ -26,3 +26,11 @@ export function isStrUrl(str = '') {
   ) && !startsWith(processedValue, 'site:');
 }
 /*eslint-enable no-mixed-operators*/
+
+export function handleRelativeURL (url) {
+  const regex =  new RegExp(/^http.*/gm);
+  if (url.match(regex)) {
+    return url;
+  }
+  return `${window.location.protocol}//${window.location.hostname}${url}`;
+}
```

Comment:
I think that there is something missing in this function.
You should also check `window.location.port`. If its 80, do thing extra but if its another port, it should add `:port` after the hostname.
This will allow supporting servers like: `http://vbanos-dev.us.archive.org:8092/web/`.